Provider/src/FirebirdSql.Data.FirebirdClient/Schema/FbSchema.cs

```diff
@@ -115,6 +118,25 @@ protected virtual string[] ParseRestrictions(string[] restrictions)
 
 		#endregion
 
+		#region Private Methods
+		/// <summary>
+		/// Determines the major version number from the Serverversion on the inner connection.
+		/// </summary>
+		/// <param name="connection">an open connection, which is used to determine the version number of the connected database server</param>
+		private void SetMajorVersionNumber(FbConnection connection)
```

Comment:
There's already `FbServerVersion.ParseServerVersion` method, that returns a `Version` type. I would store the complete `Version` and do checks on that. And also there's a `ServerVersion` property on `FbConnection` directly.

Reply:
ServerVersion isn't numeric so you can't use it for logic to determine whether you should include a meta-data table / field which is e.g. available from v3+ for instance, hence my parsing of `ServerVersion` to a numeric value :) 

You mean FbServerProperties.ParseServerVersion? Ok that could work. Do you want to store this version in InnerConnection (where Serverversion also gets its string value?). As it's now recreated every time it's needed, which might be a bit wasteful

Reply:
I think the version should be stored in `FbSchema`. The `ServerVersion` is `string` (and left raw) intentionally.